cmd/user-auth/main_test.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/cmd"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/exit"
+)
+
+func TestMain(suite *testing.M) {
+	// Run inside a func so defers will run before exit.Exit().
+	code := func() int {
+		db.PrepForTestingMain()
+		defer db.CleanupTestingMain()
+
+		return suite.Run()
+	}()
+
+	exit.Exit(code)
+}
+
+// This test ensures a CMD can start its own server.
+// It must remain or be replaced with an equivalent test if removed.
```

Comment:
I don't like this sentence.
You just say that the test needs be be here but not WHY it is important.
One of the main reasons for comments is to explain tricky things to readers,
this comment doesn't do that at all.